test/KoreBuild.Tasks.Tests/DownloadFileTests.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.IO;
+using BuildTools.Tasks.Tests;
+using Microsoft.Build.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+using Task = System.Threading.Tasks.Task;
+
+namespace KoreBuild.Tasks.Tests
+{
+    public class DownloadTests
+    {
+        private readonly ITestOutputHelper _output;
+
+        public DownloadTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact]
+        public async Task ItDownloadAFile()
+        {
+            var packages = new[]
+            {
+                new TaskItem("Newtonsoft.Json", new Hashtable { ["Version"] = "9.0.1", ["Source"] = "https://api.nuget.org/v3/index.json"} ),
+            };
+
+            var expectedPath = Path.Combine(AppContext.BaseDirectory, "microsoft.com.html");
+            var task = new DownloadFile
+            {
+                Uri = "https://microsoft.com/",
+                DestinationPath = expectedPath,
+                BuildEngine = new MockEngine(_output),
+            };
+
+            if (File.Exists(expectedPath))
+            {
+                File.Delete(expectedPath);
+            }
+
+            Assert.False(File.Exists(expectedPath), "The file should not exist yet");
+            Assert.True(await task.ExecuteAsync(), "Task should pass");
+            Assert.True(File.Exists(expectedPath), "The file should exist");
+        }
+
+        [Fact]
+        public async Task ItFailsForFilesThatDoNotExist()
+        {
+            var engine = new MockEngine(_output) { ContinueOnError = true };
+            var task = new DownloadFile
+            {
+                Uri = "http://localhost/this/file/does/not/exist",
```

Comment:
Could we add a timeout so this doesn't wait forever if there's a server responding locally?

Reply:
Added a timeout that defaults to 2 minutes to start the response, 15 minutes to finish the response.